pypeerassets/__main__.py

```diff
@@ -166,27 +168,31 @@ def deck_transfer(provider: Provider, deck: Deck,
 def find_card_transfers(provider: Provider, deck: Deck) -> Generator:
     '''find all <deck> card transfers'''
 
-    card_transfers = []
+    card_transfers = cast(list, [])
 
     if isinstance(provider, RpcNode):
+        if deck.id is None:
+            raise Exception("deck.id required to listtransactions")
         batch_data = [('getrawtransaction', [i["txid"], 1] ) for i in provider.listtransactions(deck.id)]
         result = provider.batch(batch_data)
         if result is not None:
             card_transfers = [i['result'] for i in result if result]
     else:
+        if deck.p2th_address is None:
+            raise Exception("deck.p2th_address required to listtransactions")
         if provider.listtransactions(deck.p2th_address):
-            card_transfers = (provider.getrawtransaction(i, 1) for i in
-                              provider.listtransactions(deck.p2th_address))
+            card_transfers = [provider.getrawtransaction(i, 1) for i in
```

Comment:
We use this as generator to allow for faster processing, sometimes you will required 3 card on the deck. No need to get them all, which list forces.

Reply:
gotcha, will try to give it more general type to allow for this :+1: